src/main/java/appeng/container/slot/SlotCraftingTerm.java

```diff
@@ -153,6 +157,40 @@ else if( action == InventoryAction.CRAFT_STACK ) // craft into hand, full stack
 		}
 	}
 
+	// TODO: This is really hacky and NEEDS to be solved with a full container/gui refactoring.
+	protected IRecipe findRecipe( InventoryCrafting ic, World world )
+	{
+		if( this.container instanceof ContainerCraftingTerm )
+		{
+			final ContainerCraftingTerm containerTerminal = (ContainerCraftingTerm) this.container;
+			final IRecipe recipe = containerTerminal.getCurrentRecipe();
+
+			if( recipe != null && recipe.matches( ic, world ) )
+			{
+				return containerTerminal.getCurrentRecipe();
+			}
+		}
+
+		return CraftingManager.findMatchingRecipe( ic, world );
+	}
+
+	// TODO: This is really hacky and NEEDS to be solved with a full container/gui refactoring.
+	protected NonNullList<ItemStack> getRemainingItems( InventoryCrafting ic, World world )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1161)